internal/config/cgmgr/cgroupfs_linux.go

```diff
@@ -66,6 +75,65 @@ func (m *CgroupfsManager) ContainerCgroupAbsolutePath(sbParent, containerID stri
 	return m.ContainerCgroupPath(sbParent, containerID), nil
 }
 
+// ContainerCgroupStats returns a stats object with information from the cgroup found
+// given a cgroup parent and container ID.
+func (m *CgroupfsManager) ContainerCgroupStats(sbParent, containerID string) (cgstats *CgroupStats, err error) {
+	cgMgr, err := m.GetCtrCgroupManager(sbParent, containerID)
+	if err != nil {
+		logrus.Errorf("Failed to get cgroup manager for container %s: %v", containerID, err)
+		return nil, err
+	}
+	libCtrStats, err := cgMgr.GetStats()
+	if err != nil {
+		logrus.Errorf("Failed to get cgroup stats for container %s: %v", containerID, err)
+		return nil, err
+	}
+	cgstats = &CgroupStats{
+		MostStats:     libCtrStats,
+		OtherMemStats: generateOtherMemoryStats(&libCtrStats.MemoryStats),
+		SystemNano:    time.Now().UnixNano(),
+	}
+	return cgstats, nil
+}
+
+// GetCtrCgroupManager takes the cgroup parent, and container ID.
+// It returns the raw libcontainer cgroup manager for that container.
+func (m *CgroupfsManager) GetCtrCgroupManager(sbParent, containerID string) (libctrCg.Manager, error) {
+	if !node.CgroupIsV2() {
+		if cgMgr := m.v1CtrCgMgr[containerID]; cgMgr != nil {
+			return cgMgr, nil
+		}
+	}
+	cgPath, err := m.ContainerCgroupAbsolutePath(sbParent, containerID)
+	if err != nil {
+		return nil, err
+	}
+	cgMgr, err := createCgManager(cgPath, false)
+	if err != nil {
+		return nil, err
+	}
+	if !node.CgroupIsV2() {
+		// cache only cgroup v1 managers
+		m.v1CtrCgMgr[containerID] = cgMgr
+	}
+	return cgMgr, nil
+}
+
+// RemoveCtrCgManager removes and destroys the cgroup manager for the container
+func (m *CgroupfsManager) RemoveCtrCgManager(sbParent, containerID string) {
+	cgMgr, err := m.GetCtrCgroupManager(sbParent, containerID)
```

Comment:
couldn't we just directly access the map? doesn't GetCtr... create the cgroup manager if it doesn't exist?

Reply:
To destroy the cgroup, we should get it from the map (if we use cgroup v1), or create the cgroup object on the fly for cgroup v2. `GetCtrCgroupManager` internally does this.

Reply:
FYI I don't think we should destroy the cgruop in these functions, that's done by the runtime when the container is cleaned up. We should just remove the entry from the map, if present